### PR TITLE
Rdpdr refactor server side callbacks

### PR DESCRIPTION
**Pull request by akallabeth**
* fixed a few compiler warnings
* adds additional stringify functions for later debugging
* adds `rdpdr` server side device handling, e.g. keep the device list for each peer for later reference
* adds lots of server side callbacks to intercept each part of the messages

### CODE REVIEW

File: channels/rdpdr/server/rdpdr_main.c [hunk 1]
```diff
@@ -6,6 +6,8 @@
  * Copyright 2013 Marc-Andre Moreau <marcandre.moreau@gmail.com>
  * Copyright 2015 Thincast Technologies GmbH
  * Copyright 2015 DI (FH) Martin Haimberger <martin.haimberger@thincast.com>
+ * Copyright 2022 Thincast Technologies GmbH
```
Comment (mfleisz) [inline channels/rdpdr/server/rdpdr_main.c:9]: Already in the header

**Review by mfleisz — APPROVED** [review]
Overall lgtm

File: include/freerdp/server/rdpdr.h [hunk 2]
```diff
@@ -35,6 +35,26 @@
 typedef struct s_rdpdr_server_context RdpdrServerContext;
 typedef struct s_rdpdr_server_private RdpdrServerPrivate;
 
+#pragma pack(push, 1)
+typedef struct
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply include/freerdp/server/rdpdr.h:39]: ok, done

File: channels/rdpdr/server/rdpdr_main.c (hunk 1 above)
Comment (akallabeth) [reply channels/rdpdr/server/rdpdr_main.c:9]: ok, merged

**Pull request merged**